include/natalie/symbol_object.hpp

```diff
@@ -68,21 +69,32 @@ class SymbolObject : public Object {
 
     SymbolObject(Env *env, const char *name)
         : Object { Object::Type::Symbol, GlobalEnv::the()->Symbol() }
+        , m_should_free { true }
         , m_name { strdup(name) } {
         assert(m_name);
     }
 
     SymbolObject(const char *name)
         : Object { Object::Type::Symbol, GlobalEnv::the()->Symbol() }
+        , m_should_free { true }
         , m_name { strdup(name) } {
         assert(m_name);
     }
+    // FIXME: find a better semantic for this,
+    //        maybe some sort of badge
+    SymbolObject(const char *name, size_t)
+        : Object { Object::Type::Symbol, GlobalEnv::the()->Symbol() }
+        , m_should_free { false }
+        , m_name { name } {
+        assert(m_name);
+    }
```

Comment:
Instead of using the length to signal the semantics, let's be explicit by passing a bool `owned_string`. Then we can have a single constructor:

```c++
    SymbolObject(const char *name, bool owned_string = false)
        : Object { Object::Type::Symbol, GlobalEnv::the()->Symbol() }
        , m_owned_string { owned_string }
        , m_name { owned_string ? name : strdup(name) } {
        assert(m_name);
    }
``